src/plugins/enzymes/src/EnzymesIO.cpp

```diff
@@ -133,25 +136,30 @@ EnzymeFileFormat EnzymesIO::detectFileFormat(const QString& url) {
     return EnzymeFileFormat_Unknown;
 }
 
-QList<SEnzymeData> EnzymesIO::readBairochFile(const QString& url, IOAdapterFactory* iof, TaskStateInfo& ti) {
+QList<SEnzymeData> EnzymesIO::readBairochFile(const QString& url, IOAdapterFactory* iof, U2OpStatus& os) {
     QList<SEnzymeData> res;
 
-    QScopedPointer<IOAdapter> io(iof->createIOAdapter());
+    QSharedPointer<IOAdapter> io(iof->createIOAdapter(), [](IOAdapter* ioAdapter) {
+        ioAdapter->close();
+        ioAdapter->deleteLater();
```

Comment:
Well, as far as I can see, this is as obvious as possible, I don't even know what comment I can have here, but I'll try to

Reply:
Originally it was a regular delete by a shared pointer. You changed it to a non-trivial delete. Could you please explain the change? The best place to explain is to add a comment.

Reply:
OK

Reply:
Your comment is wrong: destructors of all IOAdapters close the resource automatically: you do not need to do it manually.
My original question was why do you use `deleteLater` instead of the simple `delete`?